src/base/bittorrent/torrentimpl.cpp

```diff
@@ -2232,6 +2234,31 @@ QString TorrentImpl::createMagnetURI() const
     return QString::fromStdString(lt::make_magnet_uri(m_nativeHandle));
 }
 
+nonstd::expected<void, QString> TorrentImpl::exportToFile(const Path &path) const
+{
+    if (!hasMetadata())
+        return nonstd::make_unexpected(tr("Missing metadata"));
+
+    try
+    {
+        auto creator = lt::create_torrent(*(info().nativeInfo()));
```

Comment:
Unfortunately this won't work in almost all the cases where there is v2 torrent. You should use `lt::torrent_handle::torrent_file_with_hashes()` instead.

Reply:
I did some tests with v2 torrents and didn't found any problem. How did you test it?

Reply:
Fixed.
I think I might encountered another bug in libtorrent. I'll investigate more...